### ISSUE TITLE
hasOneResource - missing specs and questionable behaviour of store route

### ISSUE THREAD

**Issue by nialpe** [tailflow/laravel-orion#178]
Hello!

First of all thanks for this great package, it's a big time saver for me.

At the moment I am working with hasOneResource relationship controllers and I got two problems with them.

In my api routes I define a resource controller for Users and a hasOneResource controller for UserProfiles:
```php
Orion::resource('users', UserController::class)->withoutBatch();
Orion::hasOneResource('users', 'profile', UserProfileController::class)->withoutBatch();
```

First of all the specs generator doesn't create all paths for the UserProfileController. In my specs.yaml I get only:

- (POST) /api/users/{user}/profile
- (DELETE) /api/users/{user}/profile/{profile}

The _show_ and _update_ routes though are availabel - I verified this with `route:list`. They also work as the documention describes.

The second thing which feels strange is the store route for RelationControllers. My UserProfile model has a foreign id (user_id) for the User model - this foreign id is also set to be unique on database level so that a user can only have a single UserProfile. If I make two subsquent api calls to (POST) /api/users/1/profile the first call successfully creates a UserProfile, the second call though fails due to a integrity constraint violation. Shouldn't Orion's RelationController in case of 1:1 relations check in it's store route if there is already an existing relation and if so abort with some validation exception?

**Comment by alexzarbn** [comment]
Hi @nialpe,

I will check the problem with specs generation for `hasOne` relations, that wasn't reported before, thank you!

As for the `store` operation, it does, indeed, seem like a good idea to handle that on API side. Not sure though what kind of exception would be appropriate in that case 🤔

**Comment by alexzarbn** [comment]
A long overdue, but better late than never 😅

Specs generation for one-to-one relations is fixed in [v2.18.2](https://github.com/tailflow/laravel-orion/releases/tag/2.18.2) and 409 response introduced in [v2.19.0](https://github.com/tailflow/laravel-orion/releases/tag/2.19.0) to prevent storing new resources over existing one in one-to-one relation resources.
